File: languages/c/practice/pointers/soe.c
// Arrays as function arguments

#include <stdio.h>

int sum_of_elems(const int arr[], size_t len);

int main(void)
{
    int arr[] = {1, 2, 3, 4, 5};
    size_t len = sizeof(arr) / sizeof(arr[0]);

    int total = sum_of_elems(arr, len);

    printf("Sum of elements = %d\n", total);

    return 0;
}

int sum_of_elems(const int arr[], size_t len)
{
    int i, sum = 0;

    for (i = 0; i < len; i++)
    {
        sum += arr[i];
    }

    return sum;
}

File: languages/zig/practice/Pointers/soe.zig
// Arrays as function arguments

const print = @import("std").debug.print;

pub fn main() !void {
    const arr = [_]isize{ 1, 2, 3, 4, 5 };

    const total: isize = sumOfElems(&arr);

    print("Sum of elements = {}\n", .{total});
}

fn sumOfElems(arr: []const isize) isize {
    var sum: isize = 0;

    for (arr) |val| {
        sum += val;
    }

    return sum;
}
